core/src/test/java/io/questdb/griffin/engine/functions/regex/ILikeFunctionFactoryTest.java

```diff
@@ -260,6 +260,60 @@ public void testLikeStringUnderscoreAtStartAndEnd() throws Exception {
         });
     }
 
+
+    // JANE
+    @Test
+    public void testLikeEscapePercengtage() throws Exception {
+        assertMemoryLeak(() -> {
+            /**
+             * The string stored in the table is
+             * Can you give me 30% discount?
+             * The pattern I query is
+             * %30\%%
+             */
+
+
+            String sql = "create table myTable as (\n" +
+                    "select cast('Can you give me 30% discount?' as string) as comment from long_sequence(1)\n" +
+                    "union\n" +
+                    "select cast('May I get 30 USD off?' as string) as comment from long_sequence(1)\n" +
+                    "union\n" +
+                    "select cast('The target path is \\usr' as string) as comment from long_sequence(1))";
+            compiler.compile(sql, sqlExecutionContext);
+
```

Comment:
There's no need to iterate explicitly . 
Please use one of assertQuery() methods .